heareval/tasks/speech_commands.py

```diff
@@ -124,27 +125,34 @@ def requires(self):
         }
 
     @staticmethod
-    def apply_label(relative_path):
-        label = os.path.basename(os.path.dirname(relative_path))
-        if label not in WORDS and label != SILENCE:
-            label = UNKNOWN
-        return label
-
-    @staticmethod
-    def slugify_file_name(relative_path: str) -> str:
+    def unique_filestem(relpath: str) -> str:
         """
-        For speech command each speaker might have given samples for
-        different metadata. In this case, just slugifying the file name
-        without the label would cause duplicates
+        Include the label (parent directory) in the filestem.
         """
-        # Get the foldername which is the label and the filename
-        name = os.path.splitext(os.path.join(*Path(relative_path).parts[-2:]))[0]
-        return f"{slugify(str(name))}"
+        # Get the parent directory (label) and the filename
+        name = os.path.join(*Path(relpath).parts[-2:], "-")
+        # Remove the suffix
+        name = os.path.splitext(name)[0]
+        return str(name)
+
+    @staticmethod
+    def speaker_hash(unique_filestem: str) -> str:
+        """Get the speaker hash as the Split key for speech_commands"""
+        hsh = re.sub(r"-nohash-.*$", "", unique_filestem)
+        assert hsh != unique_filestem, "{unique_filestem} has no speaker hash."
```

Comment:
This supposed to be a format string? `f"{unique_filestem} ..."`

The assertion fails on the full dataset. In the full dataset it's actually `"_nohash_"`, not `"-nohash-"`, also, the background sounds have no speaker hash.

Reply:
Okay. Added the f-string and fixed the hash